prime-router/src/main/kotlin/fhirengine/translation/hl7/utils/FhirPathUtils.kt

```diff
@@ -34,59 +34,86 @@ object FhirPathUtils : Logging {
      */
     fun evaluate(
         appContext: CustomContext?,
         focusResource: Base,
         bundle: Bundle,
-        expressionNode: ExpressionNode
+        expression: String
     ): List<Base> {
-        val resources = defaultPathEngine.evaluate(appContext, focusResource, bundle, bundle, expressionNode)
-        return resources.map {
-            if (it.hasType("Reference") && (it as Reference).resource != null) {
-                it.resource as Base
-            } else it
+        val retVal = try {
+            val resources = defaultPathEngine.evaluate(appContext, focusResource, bundle, bundle, parsePath(expression))
+            resources.map {
+                if (it.hasType("Reference") && (it as Reference).resource != null) {
+                    it.resource as Base
+                } else it
+            }
+        } catch (e: Exception) {
+            // This is due to a bug in at least the extension() function
+            logger.debug(
+                "Unknown error while evaluating FHIR expression $expression. " +
+                    "Returning empty resource list.",
+                e
+            )
+            emptyList()
         }
+        logger.trace("Evaluated '$expression' to '$retVal'")
+        return retVal
     }
 
     /**
-     * Gets a boolean result from the given [expressionNode] using [bundle] and starting from a specific [focusResource].
+     * Gets a boolean result from the given [expression] using [bundle] and starting from a specific [focusResource].
      * [focusResource] can be the same as [bundle] when starting from the root.
      * [appContext] provides custom context (e.g. variables) used for the evaluation.
-     * Note that if the [expressionNode] does not evaluate to a boolean then the result is false.
+     * Note that if the [expression] does not evaluate to a boolean then the result is false.
      * @return true if the expression evaluates to true, otherwise false
      * @throws SchemaException if the FHIR path does not evaluate to a boolean type
      */
     fun evaluateCondition(
         appContext: CustomContext?,
         focusResource: Base,
         bundle: Bundle,
-        expressionNode: ExpressionNode
+        expression: String
     ): Boolean {
-        val value = defaultPathEngine.evaluate(appContext, focusResource, bundle, bundle, expressionNode)
-        return if (value.size == 1 && value[0].isBooleanPrimitive) (value[0] as BooleanType).value
-        else {
-            throw SchemaException("Condition did not evaluate to a boolean type")
+        val retVal = try {
+            val value = defaultPathEngine.evaluate(appContext, focusResource, bundle, bundle, parsePath(expression))
+            if (value.size == 1 && value[0].isBooleanPrimitive) (value[0] as BooleanType).value
+            else {
+                throw SchemaException("Condition did not evaluate to a boolean type")
```

Comment:
Should this be a `SchemaException`?  Seems like in this case the schema could be fine, but the provided bundle is the thing with the problem. BundleException?  ResourceException?

Reply:
Not in this case.  If the condition did not evaluate to a boolean it was because is was not made to be of type boolean.  E.g. someone did `Bundle.id` instead of `Bundle.id.exists()`